docs: add description of rename_XXX function (#159)
tier4/CARET_doc: docs/configuration/rename_function.md

# How to rename sub-objects in an architecture object

An architecture object has several sub-objects such as callbacks, nodes, paths, executors and topics.  

> CARET assigns names to those sub-objects to identify them. Those names come from identifiers captured at initialization tracepoints. The target application may assign different identifiers to the sub-objects on each launch or update. It prevents users from reusing the existing architecture object as it is. To enhance reusability of the architecture object, CARET serves APIs to change the names of those sub-objects.

Author: Suggestions applied.
However, a few grammar corrections may be necessary.

```suggestion
CARET assigns names to these sub-objects to identify them. These names come from identifiers captured at initialization tracepoints. The target application may assign different identifiers to the sub-objects on each launch or update. It prevents users from reusing the existing architecture object as it is. To enhance the reusability of the architecture object, CARET serves APIs to change the names of these sub-objects.
```

# How to rename element in architecture object

> Architecture objects have callback, node, path, executor, and topic objects.  

Author: Fixed it in [801f52c](https://github.com/tier4/CARET_doc/pull/159/commits/801f52c8b7166b31fd5cf29a3acdc64ddb74cdc4).

CARET serves APIs to update the names of these objects.

> The name of the target element in the architecture object is updated via the Python API.  

Reviewer: I think same meaning and it is better to delete Line 6.

Author: Fixed it in [3aa320b](https://github.com/tier4/CARET_doc/pull/159/commits/3aa320b36cd905375915e4a6cff8ef2ece6ae0ea).

Those update is also reflected in the architecture file outputted by [`arch.export`](./load_and_save.md#save).

All of the following code snippets can be executed after load environment variables with `source /path/to/ros2_caret_ws/install/setup.bash`.

## Rename callback_name

> You can update callback name from `src` to `dst` with `rename_callback` function in `Architecture` class.

Reviewer: ```suggestion
You can update callback names from `src` to `dst` with `rename_callback` function in `Architecture` class.
```
`name` is countable nouns. Other `name` must be changed.

Author: Fixed it in [d3600d3](https://github.com/tier4/CARET_doc/pull/159/commits/d3600d32534991b9eecdf156b005e361df174f6a).

> # How to update name element in architecture object

Author: Fix it in [b0e6f90](https://github.com/tier4/CARET_doc/pull/159/commits/b0e6f908c7957e0b224053578019e64ad6ac74ae).

> An architecture object loaded from a set of CTF-based record data or an architecture file has the elements callback, node, path, executor and topic.

Author: Fix it in [f24d394](https://github.com/tier4/CARET_doc/pull/159/commits/f24d3946c29deb9bd09d4a38be890e81e80e50f3).

> If you want to rename these elements, CARET serves the functions to update the name of the target element.

Author: Fix it in [f24d394](https://github.com/tier4/CARET_doc/pull/159/commits/f24d3946c29deb9bd09d4a38be890e81e80e50f3).

The name of the target element in the architecture object is updated via the Python API.

## Python API

> CARET serves Python-based APIs to update the name of the target element in an architecture object.

Author: in commit [a9307bc](https://github.com/tier4/CARET_doc/pull/159/commits/a9307bc60aa02e3c31b30781249d620334c89fa3),
Some of the elements of `## Python API` are added to the description of `# How to ...`, and `## Python API` is erased.

`### Rename ...` XXX became `## Rename ...` XX accordingly.

> All of the following code snippets can be executed after load environment variables with `source /path/to/ros2_caret_ws/install/setup.bash`.

Author: Same as above.

> All of the following code snippets can be executed after load environment variables with `source /path/to/ros2_caret_ws/install/setup.bash`.

Reviewer: If you say so.

### Update callback name

> You can update callback name in architecture object from `src` to `dst` with `rename_callback` function in `Architecture` class.

Author: Fix it in [f24d394](https://github.com/tier4/CARET_doc/pull/159/commits/f24d3946c29deb9bd09d4a38be890e81e80e50f3).

```python
# arch is caret_analyze.architecture.architecture.Architecture-based object

> 

Author: Fix it in [f24d394](https://github.com/tier4/CARET_doc/pull/159/commits/f24d3946c29deb9bd09d4a38be890e81e80e50f3).

arch.rename_callback('src', 'dst')
```

All `callback_name` in architecture object where it is `src` are updated to `dst`.

> 

Author: Fix it in [f24d394](https://github.com/tier4/CARET_doc/pull/159/commits/f24d3946c29deb9bd09d4a38be890e81e80e50f3).

This update is also reflected in the architecture file output by [`arch.export`](./load_and_save.md#save).

### Update node name

You can update node name in architecture object from `src` to `dst` with `rename_node` function in `Architecture` class.

```python
# arch is caret_analyze.architecture.architecture.Architecture-based object

> 

Author: Fix it in [f24d394](https://github.com/tier4/CARET_doc/pull/159/commits/f24d3946c29deb9bd09d4a38be890e81e80e50f3).

arch.rename_node('src', 'dst')
```

All `node_name` in architecture object where it is `src` are updated to `dst`.

> This update is also reflected in the architecture file output by [`arch.export`](./load_and_save.md#save).

Reviewer: What is 'architecture file output' ?
```suggestion
The architecture file with renamed objects is generated with an [`arch.export`](./load_and_save.md#save) command.
```

Author: Fix it in [42767a9](https://github.com/tier4/CARET_doc/pull/159/commits/42767a9fd0176d663e6af9f2587b5e2374840f6e).

This represent to architecture file **outputted** by arch.export().

Reviewer: Two my opinion.
1. I think you should use only `by` instead of `outputted`.
2. You use `update` as noun, but in other sentence, you use it as verb. It may makes confusion.
Therefore I suggest following sentence.

```
The architecture file with renamed objects is generated by the [`arch.export`](./load_and_save.md#save) command.

Author: Fixed it in [3aa320b](https://github.com/tier4/CARET_doc/pull/159/commits/3aa320b36cd905375915e4a6cff8ef2ece6ae0ea).
